TRRandomizerCore/Helpers/ItemFactory.cs

```diff
@@ -91,6 +91,36 @@ public bool CanCreateItems(string lvl, List<TR3Entity> allItems, int count, bool
         return allItems.Count + count <= _entityLimit || allowLimitBreak;
     }
 
+    public TR1Entity CreateLockedItem(string lvl, List<TR1Entity> allItems, Location location = null, bool allowLimitBreak = false)
+    {
+        TR1Entity entity = CreateItem(lvl, allItems, location, allowLimitBreak);
+        if (entity != null)
+        {
+            LockItem(lvl, allItems.IndexOf(entity));
+        }
+        return entity;
+    }
+
+    public TR2Entity CreateLockedItem(string lvl, List<TR2Entity> allItems, Location location = null, bool allowLimitBreak = false)
+    {
+        TR2Entity entity = CreateItem(lvl, allItems, location, allowLimitBreak);
+        if (entity != null)
+        {
+            LockItem(lvl, allItems.IndexOf(entity));
+        }
+        return entity;
+    }
+
+    public TR3Entity CreateLockedItem(string lvl, List<TR3Entity> allItems, Location location = null, bool allowLimitBreak = false)
+    {
+        TR3Entity entity = CreateItem(lvl, allItems, location, allowLimitBreak);
+        if (entity != null)
+        {
+            LockItem(lvl, allItems.IndexOf(entity));
+        }
+        return entity;
+    }
```

Comment:
Can we merge the 3 functions by using <T> where T : BaseTREntity or something?

Reply:
I'll rework this class, there's a lot of duplication throughout

Reply:
I've tidied up this class, so it's now completely generic. It needed a few things adjusting elsewhere too. I think I'll follow this PR up with another for setting item locations as we do it so frequently in different places e.g. instead of `entity.X = location.X; entity.Y = location.Y;` etc in so many places, just `entity.SetLocation(location);`. This one's big enough for now I think 😄